File: SoftInvPersistance/src/main/java/pe/edu/pucp/softinv/daoImp/CitaDAOImpl.java
/*
package pe.edu.pucp.softinv.daoImp;

import pe.edu.pucp.softinv.dao.CitaDAO;
import pe.edu.pucp.softinv.dao.ClienteDAO;
import pe.edu.pucp.softinv.dao.EmpleadoDAO;
import pe.edu.pucp.softinv.dao.ServicioDAO;
import pe.edu.pucp.softinv.daoImp.util.Columna;
import pe.edu.pucp.softinv.model.Cita.CitaDTO;
import pe.edu.pucp.softinv.model.Personas.ClienteDTO;
import pe.edu.pucp.softinv.model.Personas.EmpleadoDTO;

import java.sql.SQLException;
import java.util.ArrayList;
import java.util.Date;

public class CitaDAOImpl extends DAOImplBase implements CitaDAO {

    private CitaDTO cita;

    public CitaDAOImpl() {
        super("CITAS");
        cita = null;
        retornarLlavePrimaria=true;
    }

    @Override
    protected void configurarListaDeColumnas() {
        listaColumnas.add(new Columna("CITA_ID", true, true));
        listaColumnas.add(new Columna("EMPLEADO_ID", false, false));
        listaColumnas.add(new Columna("CLIENTE_ID", false, false));
        listaColumnas.add(new Columna("SERVICIO_ID", false, false));
        listaColumnas.add(new Columna("FECHA", false, false));
        listaColumnas.add(new Columna("HORA_INICIO", false, false));
        listaColumnas.add(new Columna("HORA_FIN", false, false));
    }

    @Override
    public Integer insertar(CitaDTO cita) {
        this.cita = cita;
        return super.insertar();
    }

    @Override
    public Integer modificar(CitaDTO cita) {
        this.cita = cita;
        return super.modificar();
    }

    @Override
    public Integer eliminar(CitaDTO cita) {
        this.cita = cita;
        return super.eliminar();
    }

    @Override
    ArrayList<CitaDTO> listarCitasPorPeriodo(Date inicio, Date fin) {
        ArrayList<CitaDTO> citas = new ArrayList<>();
        String sql = "SELECT * FROM CITAS";
        this.abrirConexion();
        try {
            this.colocarSQLEnStatement(sql);
            this.ejecutarSelectEnDB();
            while (this.resultSet.next()) {
                Date fecha=statement.getDate("FECHA");
                if (fecha.compareTo(inicio) >= 0 && fecha.compareTo(fin) <= 0){
                    int empleadoId, clienteId, servicioId;
                    CitaDTO cita = new CitaDTO();
                    EmpleadoDAO empleado = new EmpleadoDAOImpl();
                    ClienteDAO cliente = new ClienteDAOimpl();
                    ServicioDAO servicio = new ServicioDAOImpl();

                    cita.setId(resultSet.getInt("CITA_ID"));
                    empleadoId= resultSet.getInt("EMPLEADO_ID");
                    cita.setEmpleado(empleado.obtenerPorId(empleadoId));
                    clienteId= resultSet.getInt("CLIENTE_ID");
                    cita.setCliente(cliente.obtenerPorId(clienteId));
                    servicioId= resultSet.getInt("SERVICIO_ID");
                    cita.setServicio(servicio.obtenerPorId(servicioId));
                    cita.setFecha(resultSet.getDate("FECHA"));
                    cita.setHoraIni(resultSet.getTime("HORA_INICIO"));
                    cita.setHoraFin(resultSet.getTime("HORA_FIN"));

                    citas.add(cita);
                }
            }
        } catch (SQLException ex) {
            throw new RuntimeException(ex);
        }
            return citas;
        }
    }
}
*/

File: C#/SoftInvPersistance/DAOImpl/CitaDAOImpl.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Data.Common;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Linq;
using softinv.model;
using SoftInv.DAO;
using SoftInv.DAOImpl.Util;
using static System.Collections.Specialized.BitVector32;

namespace SoftInv.DAOImpl
{
    public class CitaDAOImpl: DAOImplBase, CitaDAO {

        private CitaDTO cita;

        public CitaDAOImpl(): base("CITAS"){
            cita = null;
            retornarLlavePrimaria=true;
        }

        protected override void ConfigurarListaDeColumnas() {
            listaColumnas.Add(new Columna("CITA_ID", true, true));
            listaColumnas.Add(new Columna("EMPLEADO_ID", false, false));
            listaColumnas.Add(new Columna("CLIENTE_ID", false, false));
            listaColumnas.Add(new Columna("SERVICIO_ID", false, false));
            listaColumnas.Add(new Columna("FECHA", false, false));
            listaColumnas.Add(new Columna("HORA_INICIO", false, false));
            listaColumnas.Add(new Columna("HORA_FIN", false, false));
        }
        public int Insertar(CitaDTO cita) {
            this.cita = cita;
            return base.Insertar();
        }

        public int modificar(CitaDTO cita) {
            this.cita = cita;
            return base.Modificar();
        }

        public int eliminar(CitaDTO cita) {
            this.cita = cita;
            return base.Eliminar();
        }

        IList<CitaDTO> listarCitasPorPeriodo(DateTime inicio, DateTime fin) {
            IList<CitaDTO> citas = new IList<>();
            String sql = "SELECT * FROM CITAS";
            this.AbrirConexion();
            try {
                this.ColocarSQLenComando(sql);
                this.EjecutarConsultaEnBD();
                while (this.resultSet.next()) {

                    if (resultSet.getDate("FECHA")>=inicio and resultSet.getDate("FECHA")<=fin){

                        CitaDTO cita = new Cita();
                        cita.setId(resultset.GEtIn("CITA_ID");
                        cita.setEmpleado(resultset.GEtIn("EMPLEADO_ID");
                        cita.setClienteId(resultset.GEtIn("CLIENTE_ID");
                        cita.setServicioId(resultset.GEtIn("SERVICIO_ID");
                        cita.setFecha(resultset.GEtIn("FECHA");
                        cita.setHoraIni(resultset.GEtIn("HORA_INICIO");
                        cita.setHoraFin(resultset.GEtIn("HORA_FIN");

                        citas.Add(cita);
                    }
                }
            } catch (SQLException ex) {
                throw new RuntimeException(ex);
            }
                return citas;
            }
        }
}